### PR TITLE
FIX_md_parser: v2 

### LINKED ISSUE #283: revisit behavior of import_sample_info
mongo doesn't allow `.` in the key field so we can not  have metadata like:

`{'sample_composition':{'0.8mm_glass_capillary': 1}}`

need to rethink the behavior carefully.

### PR DESCRIPTION
**Pull request by chiahaoliu**
closes #283, #350 

This PR summarize suggested logics mentioned in the close PR #344.
It includes fixes on two parts:
1. fundamental fix for md_parser logic so that we won't encounter the same issue currently mentioned in issue #283. Therefore our example sheet doesn't need to be updated.
2. add validation option to ``import_sample_info`` function. User can use it to inspect metadata entered inside the spreadsheet.
3. add mutation logic when instantiating ``Sample`` so if user is not aware of invalid ``.`` in key fields, program shows warning(but not error) and mutates the metadata automatically to make experiments running.

This PR is ready for review, thanks.

### CODE REVIEW

**Comment by chiahaoliu** [comment]
unknown error appears. close it for now and work on it tomorrow

**Comment by chiahaoliu** [comment]
error comes from git confusion happened in rebase. Solved and reopen PR

File: xpdacq/utils.py [hunk 1]
```diff
@@ -372,18 +360,26 @@ def create_yaml(self, bt):
...
-                         for el in self.parsed_sa_md_list if
-                         el['sample_name'].startswith('bkgd')]
-        for bkg_name in set(bkg_name_list):
-            bkg_dict = {'sample_name': bkg_name,
-                        'sample_composition': {bkg_name: 1},
-                        'is_background': True}
-            Sample(bt, bkg_dict)
+        for md_dict in self.parsed_sa_md_list:
+            sample_name = md_dict.get('sample_name')
+            if sample_name.startswith('bkgd'):
```
Comment (CJ-Wright) [inline xpdacq/utils.py]: Is there a better way to be more explicit about this? (eg, have a field in the spreadsheet stating that a sample is a background?)

**Comment by chiahaoliu** [comment]
close based on the discussion with @CJ-Wright  and @sbillinge. Replace with new PR

**Pull request closed without merging**